spec/lib/active_merchant/billing/gateways/shopify_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'spec_helper'
+require 'active_merchant/billing/gateways/shopify'
+
+module ActiveMerchant::Billing
+  RSpec.describe ShopifyGateway do
+    let(:api_key) { 'api_key' }
+    let(:password) { 'password' }
+    let(:shop_name) { 'shop_name' }
+
+    let(:gateway) do
+      described_class.new(api_key: api_key,
+                          password: password,
+                          shop_name: shop_name)
+    end
+
+    around(:each) do |example|
+      begin
+        previously = ShopifyAPI::Base.site
+        example.run
+      ensure
+        ShopifyAPI::Base.site = previously
+      end
+    end
+
+    describe "#void" do
+      let(:transaction_id) { '123' }
+      let(:transaction_amount) { 1 }
+      let(:transaction) { double(:transaction, amount: transaction_amount, id: transaction_id) }
+
+      let(:refunder) { double(:refunder, perform: true) }
+
+      before do
+        allow(ShopifyAPI::Transaction).to receive(:find).and_return(transaction)
+        allow(ShopifyRefunder).to receive(:new).and_return(refunder)
+      end
+
+      it "performs a refund" do
+        expect(refunder).to receive(:perform).once
+
+        gateway.void(transaction_id, { order_id: transaction_id })
+      end
+
+      context "when the transaction can't be found" do
+        let(:transaction) { nil }
+
+        it "raises an exception" do
+          cause = ->{ gateway.void(transaction_id, { order_id: transaction_id }) }
+
+          expect(&cause).to raise_error ::ActiveMerchant::Billing::ShopifyGateway::TransactionNotFoundError
+        end
+      end
+    end
+
+    describe "#refund" do
```

Comment:
Can this be dramatically pared down because these behaviours have been delegated to ShopifyRefunder?